tests/mutation_test.cc

```diff
@@ -1320,6 +1320,99 @@ SEASTAR_THREAD_TEST_CASE(test_mutation_upgrade_type_change) {
     assert_that(m).is_equal_to(m2);
 }
 
+SEASTAR_THREAD_TEST_CASE(test_row_marker) {
+    auto ttl = std::chrono::seconds(1);
+    auto t1 = gc_clock::now();
+    auto t0 = t1 - std::chrono::seconds(1);
+    auto t2 = t1 + std::chrono::seconds(1);
+    auto t3 = t2 + std::chrono::seconds(1);
+
+    can_gc_fn never_gc = [] (tombstone) { return false; };
+
+    auto must_be_alive = [&] (row_marker mark, gc_clock::time_point t) {
+        BOOST_TEST_MESSAGE(format("must_be_alive({}, {})", mark, t));
+        BOOST_REQUIRE(mark.is_live(tombstone(), t));
+        BOOST_REQUIRE(mark.is_missing() || !mark.is_dead(t));
+        BOOST_REQUIRE(mark.compact_and_expire(tombstone(), t, never_gc, gc_clock::time_point()));
+    };
+
+    auto must_be_dead = [&] (row_marker mark, gc_clock::time_point t) {
+        BOOST_TEST_MESSAGE(format("must_be_dead({}, {})", mark, t));
+        BOOST_REQUIRE(!mark.is_live(tombstone(), t));
+        BOOST_REQUIRE(mark.is_missing() || mark.is_dead(t));
+        BOOST_REQUIRE(!mark.compact_and_expire(tombstone(), t, never_gc, gc_clock::time_point()));
+    };
+
+    // Without timestamp it is never alive
+    row_marker m1;
+    must_be_dead(m1, t0);
+    must_be_dead(m1, t1);
+    must_be_dead(m1, t2);
+    must_be_dead(m1, t3);
+
+    // Without ttl it is always alive
+    row_marker m2(api::new_timestamp());
```

Comment:
is api::new_timestamp() supposed to be the same as t1?
The reason I'm asking is that not a good property of the test that it slightly depends on how much time passed between t1 and right now. Could we just use t1 here?

Reply:
`t1` cannot be used here, `gc_clock::time_point` is a different type from `api::timestamp_type`.

Actually, the logic that this test verifies does not take the timestamp into consideration. I changed values used in this test into one constant. I initially used `api::new_timestamp` because it was also used in other tests.